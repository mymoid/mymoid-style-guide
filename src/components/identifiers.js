export const AUTOCOMPLETE = 'MYMOAutocomplete';
export const APP_BAR = 'MYMOAppBar';
export const AVATAR = 'MYMOAvatar';
export const BUTTON = 'MYMOButton';
export const CHECKBOX = 'MYMOCheckbox';
export const CHIP = 'MYMOChip';
export const DIALOG = 'MYMODialog';
export const DRAWER = 'MYMODrawer';
export const DROPDOWN = 'MYMODropdown';
export const ERROR = 'MYMOError';
export const INPUT = 'MYMOInput';
export const LAYOUT = 'MYMOLayout';
export const LINK = 'MYMOLink';
export const MEDIA = 'MYMOMedia';
export const MENU = 'MYMOMenu';
export const NAVIGATION = 'MYMONavigation';
export const OTP = 'MYMOOtp';
export const OVERLAY = 'MYMOOverlay';
export const RIPPLE = 'MYMORipple';
export const CARDINPUT = 'MYMOSCardInput';
export const SPINNER = 'MYMOSpinner';
export const STEPS = 'MYMOSteps';
export const SWITCH = 'MYMOSwitch';
export const TABLE = 'MYMOTable';
export const TABS = 'MYMOTabs';
