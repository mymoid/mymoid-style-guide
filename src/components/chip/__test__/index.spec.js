import React from 'react';
import Enzyme, { mount } from 'enzyme';
import { themr } from 'react-css-themr';
import { CHIP } from '../../identifiers';
import { chipFactory } from '../Chip';
import Adapter from 'enzyme-adapter-react-16';
import { mountToJson } from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });

const Avatar = ({ title }) => <span>{title}</span>; // eslint-disable-line react/prop-types
const Chip = themr(CHIP)(chipFactory(Avatar));

describe('Chip', () => {
  describe('with avatar', () => {
    it('adds the avatar class to the element', () => {
      const wrapper = mount(
        <Chip theme={{ avatar: 'avatar-class' }}>
          <Avatar title="Test" />
          <span>Test</span>
        </Chip>
      );
      expect(mountToJson(wrapper)).toMatchSnapshot();
    });

    describe('without avatar', () => {
      it('does not add avatar class to the element', () => {
        const wrapper = mount(
          <Chip theme={{ avatar: 'avatar-class' }}>
            <span>Test</span>
          </Chip>
        );
        expect(mountToJson(wrapper)).toMatchSnapshot();
      });
    });
  });
});
