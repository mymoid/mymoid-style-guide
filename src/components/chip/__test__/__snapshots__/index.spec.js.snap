// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chip with avatar adds the avatar class to the element 1`] = `
<ThemedChip
  className=""
  composeTheme="deeply"
  deletable={false}
  mapThemrProps={[Function]}
  theme={
    Object {
      "avatar": "avatar-class",
    }
  }
>
  <Chip
    className=""
    deletable={false}
    theme={
      Object {
        "avatar": "avatar-class",
      }
    }
  >
    <div
      className="avatar-class"
      data-mymoid="chip"
    >
      <Avatar
        title="Test"
      >
        <span>
          Test
        </span>
      </Avatar>
      <span>
        Test
      </span>
    </div>
  </Chip>
</ThemedChip>
`;

exports[`Chip with avatar without avatar does not add avatar class to the element 1`] = `
<ThemedChip
  className=""
  composeTheme="deeply"
  deletable={false}
  mapThemrProps={[Function]}
  theme={
    Object {
      "avatar": "avatar-class",
    }
  }
>
  <Chip
    className=""
    deletable={false}
    theme={
      Object {
        "avatar": "avatar-class",
      }
    }
  >
    <div
      className=""
      data-mymoid="chip"
    >
      <span>
        Test
      </span>
    </div>
  </Chip>
</ThemedChip>
`;
